superset-frontend/src/features/alerts/buildErrorTooltipMessage.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { styled } from '@superset-ui/core';
+import { ValidationObject } from './types';
+import { TRANSLATIONS } from './AlertReportModal';
+
+const StyledList = styled.ul`
+  margin-left: ${({ theme }) => theme.gridUnit * 2}px;
+  padding-inline-start: ${({ theme }) => theme.gridUnit * 3}px;
+`;
+
+export const buildErrorTooltipMessage = (
```

Comment:
1. Is `build` arg necessary? We could move the `hasError` logic from `AlertReportModal` in here
2. I'd suggest returning the error message instead of passing the set function as arg. That would also allow us to write unit tests for it.
3. Can we add some unit tests? 🙂

Reply:
changes have been addressed